### PR TITLE
consistent handling of filled reflections

### PR DESCRIPTION
**Pull request by tjlane**
@alisiafadini noticed that the final reported negentropy from a script run was higher than if re-computed from the final written MTZ, issue #57.

We discovered this is because the negentropy values reported during algorithm interactions include reflections that are filled up to the highest resolution, while the final written MTZ retains exactly the same HLK indices as the input MTZ.

We elected to report an MTZ that includes these filled reflections, but more work is needed to understand this behavior in detail, see #58.

### CODE REVIEW

File: meteor/diffmaps.py [hunk 1]
```diff
@@ -7,28 +7,13 @@
...
 import reciprocalspaceship as rs
 
-from .rsmap import Map, _assert_is_map
-from .settings import DEFAULT_KPARAMS_TO_SCAN, MAP_SAMPLING
-from .utils import filter_common_indices
-from .validate import ScalarMaximizer, negentropy
+from .rsmap import Map, assert_is_map
+from .settings import DEFAULT_KPARAMS_TO_SCAN
+from .utils import assert_isomorphous, filter_common_indices
+from .validate import ScalarMaximizer, map_negentropy
```
Comment (alisiafadini) [inline meteor/diffmaps.py:13]: this is a very useful wrapper :)

File: meteor/rsmap.py [hunk 2]
```diff
@@ -295,30 +309,86 @@ def canonicalize_amplitudes(self) -> None:
...
+        complex_structurefactor: rs.DataSeries,
+        *,
+        index: None = None,
+        cell: CellType | None = None,
+        spacegroup: SpacegroupType | None = None,
+    ) -> Map: ...
+
+    @overload
+    @classmethod
+    def from_structurefactor(
```
Comment (alisiafadini) [inline meteor/rsmap.py:328]: maybe this name is a little too vague. Not clear that structurefactor is an `rs.DataSeries`.

File: meteor/utils.py [hunk 3]
```diff
@@ -20,6 +19,17 @@
 class ShapeMismatchError(Exception): ...
 
 
+class NotIsomorphousError(RuntimeError): ...
+
+
+def assert_isomorphous(*, derivative: rs.DataSet, native: rs.DataSet) -> None:
+    if not native.is_isomorphous(derivative):
```
Comment (alisiafadini) [inline meteor/utils.py:26]: is this coming from rs_spaceship?

**Review by alisiafadini — APPROVED** [review]

File: meteor/utils.py (hunk 3 above)
Comment (tjlane) [reply meteor/utils.py:26]: yes, it's a method on `rs.DataSet`s

File: meteor/rsmap.py (hunk 2 above)
Comment (tjlane) [reply meteor/rsmap.py:328]: I tend to agree, but this is overwriting a method from `reciprocalspaceship`, so I would vote we keep it consistent.

https://github.com/rs-station/reciprocalspaceship/blob/cbf30c1c8438326baddf6a9898c2f5b5be3f7b43/reciprocalspaceship/dataset.py#L484-L528

**Pull request merged**